src/black/__init__.py

```diff
@@ -2623,7 +2622,7 @@ def init_st(ST: Type[StringTransformer]) -> StringTransformer:
     transformers: List[Transformer]
     if (
         not line.contains_uncollapsable_type_comments()
-        and not line.should_explode
+        and not line.should_explode  # XXX suspicious
```

Comment:
This is probably the line that causes instability but it cannot be easily removed.